### PR TITLE
AORC BMI unit test for model engine team

### PR DESCRIPTION
**Pull request by jduckerOWP**
Instructions to execute the AORC BMI module are located in the "README.md" file. The "requirements.txt" file highlight the required python packages needed to properly execute the AORC BMI module. Currently, this AORC BMI unit module dynamically downloads AORC data from the ERRDAP server during the model execution. Therefore, the user will need to test this script on a Linux environment that is capable of connecting to the server.

[Short description explaining the high-level reason for the pull request]

## Additions

-

## Removals

-

## Changes

-

## Testing

1. Follow Instructions within the README.md file.  

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Linux

### CODE REVIEW

File: AORC_BMI_Unit_Test/AORC_model.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        filename = wget.download(AORC_url_link,bar=None)
+
+        # load AORC netcdf file
+        AORC_ncfile = nc.Dataset(AORC_file)
+
+        # Send regridded AORC data back to model output
+        i = 0
+        for var in AORC_met_vars:           
+            met_data = np.where(AORC_ncfile.variables[var][:].data == missing_value[i], np.nan, AORC_ncfile.variables[var][:].data) 
+            print(met_data)
```
Comment (jduckerOWP) [inline AORC_BMI_Unit_Test/AORC_model.py:56]: This print statement was a flag for looking at NAN data previously. This has been fixed and is no longer required. Should be deleted to reduce print statement output.

File: AORC_BMI_Unit_Test/AORC_bmi_model.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+    def get_var_location(self, name):
+        
+        if name in (self._output_var_names + self._input_var_names):
+            return self._var_loc
+
+    #-------------------------------------------------------------------
+    def get_var_rank(self, long_var_name):
+
+        return np.int16(0)
+
```
Comment (hellkite500) [inline AORC_BMI_Unit_Test/AORC_bmi_model.py:343]: Int16 isn't ngen compatible at the moment, use int32

File: AORC_BMI_Unit_Test/AORC_bmi_model.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+        ----------
+        var_name : str
+            Name of model variable for which to set values.
+        indices : array_like
+            Array of indices of the variable into which analogous provided values should be set.
+        src : array_like
+            Array of new values.
+        """
+        # This is not particularly efficient, but it is functionally correct.
+        for i in range(indices.shape[0]):
```
Comment (hellkite500) [inline AORC_BMI_Unit_Test/AORC_bmi_model.py:399]: See previous comment on set value

**Review by hellkite500 — COMMENTED** [review]
A couple notes from initial look. I'll get a more thorough review in the near future.

File: AORC_BMI_Unit_Test/AORC_bmi_model.py [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+        self.cfg_bmi = self._parse_config(cfg)
+
+
+    #-------------- Initalize AORC Forcings specifications for NextGen -----#
+        # Set attributes that relate to standard
+        # AORC url links for grabbing data from ERRDAP server
+        self._aorc_beg = "AORC-OWP_"
+        self._aorc_end = "z.nc4"
+        self._aorc_new_end = ".nc4"
+        self._base_url = 'https://nwcal-wrds-ti01.nwc.nws.noaa.gov/aorc_erddap/erddap/files/erddap_dadf_1012_a549/'
```
[the remark this answers is not in the document]
Comment (hellkite500) [reply AORC_BMI_Unit_Test/AORC_bmi_model.py:112]: Rebase this from the history

**Comment by hellkite500** [comment]
Closing in favor of #12

**Pull request closed without merging**